scripts/security/gen-test-images.sh

```
@@ -0,0 +1,42 @@
+  local hex="$1"
+  local output="$2"
+  # Use python3 as a reliable hex-to-bin converter if xxd is missing,
+  # but we want to stick to shell if possible.
+  # However, printf '\xHH' is the most standard shell way.
+
+  # Prepare the printf string: 89 -> \x89
+  local fmt=""
+  for ((i = 0; i < ${#hex}; i += 2)); do
+    fmt="${fmt}\\x${hex:$i:2}"
+  done
+  printf "${fmt}" >"${output}"
```

Shellcheck (0.11.0)

[info] Don't use variables in the printf format string. Use printf '..%s..' "$foo".

(SC2059)